test/dynamo/test_trace_rules.py

```diff
@@ -108,195 +73,11 @@ def get_func_inlinelist():
-
-                if isinstance(obj, torch._ops.HigherOrderOperator):
-                    continue
-
-                # We want to trace through `grad` and `vmap`
-                if obj in (
-                    torch.func.grad,
-                    deprecated_func.grad,
-                    torch.func.vmap,
-                    deprecated_func.vmap,
-                    torch.nn.functional.triplet_margin_with_distance_loss,
-                    torch.cond,
-                ):
-                    continue
-
-                if isinstance(obj, types.ModuleType):
-                    if obj.__name__.startswith("torch.") and _is_allowed_module_prefix(
-                        obj
-                    ):
-                        torch_object_ids[id(obj)] = f"{module.__name__}.{name}"
-                        _find_torch_objects(obj)
-                elif _is_allowed_module_prefix(obj):
-                    torch_object_ids[id(obj)] = f"{module.__name__}.{name}"
-                    if (
-                        issubclass(type(obj), type)
-                        and "__enter__" in obj.__dict__
-                        and "__exit__" in obj.__dict__
-                    ):
-                        torch_objects.add(obj)
-                elif inspect.getmodule(obj) is None and not is_safe_constant(obj):
-                    torch_object_ids[id(obj)] = f"{module.__name__}.{name}"
-                    if (
-                        issubclass(type(obj), type)
-                        and "__enter__" in obj.__dict__
-                        and "__exit__" in obj.__dict__
-                    ):
-                        torch_objects.add(obj)
-
-    _find_torch_objects(torch)
-    _find_torch_objects(math)
-
-    if config.trace_distributed:
-        from torch.distributed import _functional_collectives_impl as fci
-
-        for f in [
-            fci._all_gather_into_tensor,
-            fci._all_reduce,
-            fci._reduce_scatter_tensor,
-            fci._all_reduce_coalesced,
-            fci._all_gather_into_tensor_coalesced,
-            fci._reduce_scatter_tensor_coalesced,
-        ]:
-            torch_object_ids[id(f)] = repr(f)
-
-    # torch.Tensor.{fn}
-    for name in dir(torch.Tensor):
-        method = getattr(torch.Tensor, name)
-        if isinstance(
-            method, (types.MethodDescriptorType, types.WrapperDescriptorType)
-        ):
-            torch_object_ids[id(method)] = f"torch.Tensor.{name}"
-
-    for idx in _disallowed_function_ids():
-        if idx in torch_object_ids:
-            del torch_object_ids[idx]
-
-    for extra in (is_fx_tracing, is_compiling):
-        torch_object_ids[id(extra)] = f"{extra.__module__}.{extra.__name__}"
-
-    return torch_objects
+# Generate the allowed objects based on heuristic defined in `allowed_functions.py`,
+# allowed objects include:
+# - Torch context manager classes.
+def generate_allowed_object_set():
+    return gen_allowed_objs_and_ids()[1]
```

Comment:
The ```gen_allowed_objs_and_ids``` was duplicated in my last PR, to save the sync cost, I removed one and import from the other one.